galpy/potential/TwoPowerSphericalPotential.py

```diff
@@ -53,30 +56,63 @@ def __init__(self,amp=1.,a=5.,alpha=1.5,beta=3.5,normalize=False,
         HISTORY:
 
            2010-07-09 - Started - Bovy (NYU)
+           2019-10-20 - Updated - Starkman (UofT)
 
         """
-        if alpha == round(alpha) and beta == round(beta):
-            Potential.__init__(self,amp=amp,ro=ro,vo=vo,amp_units='mass')
-            integerSelf= TwoPowerIntegerSphericalPotential(amp=1.,a=a,
-                                                           alpha=int(alpha),
-                                                           beta=int(beta),
-                                                           normalize=False)
-            self.integerSelf= integerSelf
+        # instantiate
+        super(TwoPowerSphericalPotential, self).__init__(amp=amp,ro=ro,vo=vo,amp_units='mass')
+        # integerSelf
+        if ((self.__class__ == TwoPowerSphericalPotential) &
+            (alpha == round(alpha)) & (beta == round(beta))):
+            self.integerSelf= TwoPowerIntegerSphericalPotential(
+                amp=1.,a=a,alpha=round(alpha),beta=round(beta),
+                normalize=False,ro=None,vo=None)
         else:
-            Potential.__init__(self,amp=amp,ro=ro,vo=vo,amp_units='mass')
             self.integerSelf= None
+        # correcting quantities
         if _APY_LOADED and isinstance(a,units.Quantity):
-            a= a.to(units.kpc).value/self._ro
+            a= a.to_value(units.kpc)/self._ro
+        # setting properties
         self.a= a
         self._scale= self.a
-        self.alpha= alpha
-        self.beta= beta
+        self._alpha= alpha
+        self._beta= beta
+        # normalizing
         if normalize or \
                 (isinstance(normalize,(int,float)) \
                      and not isinstance(normalize,bool)): #pragma: no cover
             self.normalize(normalize)
         return None
 
+    @property
+    def alpha(self):
+        return self._alpha
+    @alpha.setter
+    def alpha(self, alpha):
+        # checking if alpha allowed to change
+        self._alpha = alpha  # setting alpha
+        # update .integerSelf
+        if alpha == round(alpha) and self._beta == round(self._beta):
+            self.integerSelf= TwoPowerIntegerSphericalPotential(
+                amp=1.,a=self.a,alpha=round(alpha),beta=round(self._beta),
+                normalize=False)
+        else:
+            self.integerSelf= None
+
+    @property
+    def beta(self):
+        return self._beta
+    @beta.setter
+    def beta(self, beta):
+        self._beta = beta  # setting alpha
+        # update .integerSelf
+        if self._alpha == round(self._alpha) and beta == round(beta):
+            self.integerSelf= TwoPowerIntegerSphericalPotential(
+                amp=1.,a=self.a,alpha=round(self._alpha),beta=round(beta),
+                normalize=False)
+        else:
+            self.integerSelf= None
```

Comment:
I'm not sure I see the point of allowing alpha and beta to be adjusted like this, this isn't something that is typically allowed for galpy potentials, so introduces inconsistencies in how the parameters of potentials can be changed. I'd prefer to remove this part of the PR and perhaps re-think in a more general way in the future

Reply:
done.